### PR TITLE
React version up

### PR DESCRIPTION
**Pull request by hachuu**
# React 16.13.1 to 17.0.1
- yarn upgrade react@17.0.1 react-dom@17.0.1
- yarn upgrade -D @babel/{core,preset-{env,react}}
- yarn add -D @babel/plugin-transform-react-jsx
- .babelrc 수정

### CODE REVIEW

**Comment by eomttt** [comment]
제목 수정해주세요

File: .babelrc [hunk 1]
```diff
@@ -1,7 +1,9 @@
 {
  "presets": [
    "@babel/preset-env",
-   "@babel/preset-react",
+   ["@babel/preset-react", {
+    "runtime": "automatic"
```
Comment (eomttt) [inline .babelrc:5]: 이건 뭐임??

File: package.json [hunk 2]
```diff
@@ -23,14 +23,15 @@
...
+    "react": "^17.0.1",
     "react-apollo": "^3.1.5",
-    "react-dom": "^16.13.1",
+    "react-dom": "^17.0.1",
     "react-router-dom": "^5.1.2",
     "react-use": "^15.3.4"
   },
   "devDependencies": {
     "@babel/core": "^7.9.6",
+    "@babel/plugin-transform-react-jsx": "^7.12.7",
```
Comment (eomttt) [inline package.json:34]: 이것도 업뎃 해야함?

File: src/containers/SearchedMovies/index.tsx [hunk 3]
```diff
@@ -32,7 +32,6 @@ const SearchedMoviesContainer = () => {
         (error) => {
           if (error.code === 1) {
             alert('위치 정보를 가져 올 수 있도록 허용해주세요.');
```
Comment (eomttt) [inline src/containers/SearchedMovies/index.tsx:34]: return 추가해야할듯

File: src/containers/TheaterTimeTable/index.tsx [hunk 4]
```diff
@@ -10,7 +10,7 @@ import { ITimeTableData } from 'interfaces/timeTable';
 
 interface TheaterTimeTableProps {
   theaterInfo: TheaterInfo;
-  onSetMovies?: (movieCard: ISearchedMovieCard[]) => void;
+  onSetMovies: (movieCard: ISearchedMovieCard[]) => void;
```
Comment (eomttt) [inline src/containers/TheaterTimeTable/index.tsx:13]: 이거 왜 린트 에러난거지?

File: .babelrc (hunk 1 above)
Comment (hachuu) [reply .babelrc:5]: https://marlom.dev/upgrade-to-react-17-and-webpack-5
이거 보고 업데이트하다가 추가했음

File: package.json (hunk 2 above)
Comment (hachuu) [reply package.json:34]: https://marlom.dev/upgrade-to-react-17-and-webpack-5
이거 보고 업데이트하다가 추가했음

Comment (hachuu) [reply package.json:34]: https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html

File: src/containers/TheaterTimeTable/index.tsx (hunk 4 above)
Comment (hachuu) [reply src/containers/TheaterTimeTable/index.tsx:13]: propType "onSetMovies" is not required, but has no corresponding defaultProps declaration  react/require-default-props

File: package.json (hunk 2 above)
Comment (hachuu) [reply package.json:34]: https://babeljs.io/docs/en/babel-plugin-transform-react-jsx

**Review by eomttt — APPROVED** [review]

**Pull request merged**